src/main/java/cz/datadriven/utils/config/view/ConfigViewFactory.java

```diff
@@ -42,22 +55,32 @@
    * @param <T> type of the view class to be created
    * @return the view
    */
-  @SuppressWarnings("unchecked")
   public static <T> T create(Class<T> configViewClass, Config config) {
     if (!ConfigViewProxy.canProxy(configViewClass)) {
       throw new IllegalArgumentException(
-          "Can not instantiate ConfigView for "
-              + "class [ "
-              + configViewClass
-              + " ]. Did you forget @ConfigView annotation?");
+          String.format(
+              "Can not instantiate ConfigView for class [%s]. Did you forget @ConfigView annotation?",
+              configViewClass));
     }
-    final Enhancer enhancer = new Enhancer();
-    enhancer.setCallback(new ConfigViewProxy(new ConfigViewProxy.Factory(config)));
-    if (configViewClass.isInterface()) {
-      enhancer.setInterfaces(new Class[] {configViewClass});
-    } else {
-      enhancer.setSuperclass(configViewClass);
+    try {
+      return new ByteBuddy(ClassFileVersion.JAVA_V8)
+          .subclass(configViewClass)
+          .method(
+              ElementMatchers.isAnnotatedWith(ANNOTATION_TYPE_DESCRIPTORS::contains)
+                  .or(ElementMatchers.isDeclaredBy(RawConfigAware.class)))
+          .intercept(
+              InvocationHandlerAdapter.of(new ConfigViewProxy(new ConfigViewProxy.Factory(config))))
+          .make()
+          .load(ConfigViewFactory.class.getClassLoader(), ClassLoadingStrategy.Default.INJECTION)
+          .getLoaded()
+          .getDeclaredConstructor()
+          .newInstance();
+    } catch (InstantiationException
+        | InvocationTargetException
+        | NoSuchMethodException
+        | IllegalAccessException e) {
+      throw new RuntimeException(
```

Comment:
Some more specific implementation of `RuntimeException` should be used. For example sonarqube considers instantiating `RuntimeException` a bug.

Reply:
switched to illegal state